packages/typespec-vs/src/VSExtension.cs

```diff
@@ -144,9 +147,41 @@ public async Task OnLoadedAsync()
 
         public Task OnServerInitializedAsync()
         {
+            string typespecCompilerVersion = GetTypespecVersion();
+            if (_telemetryService != null && !string.IsNullOrWhiteSpace(typespecCompilerVersion))
+            {
+                IVsTelemetryEvent telemetryEvent = _telemetryService.CreateEvent("vs/typespec/vs-extension/start-extension");
+                telemetryEvent.SetStringProperty("vs.typespec.compiler.version", typespecCompilerVersion);
+                _telemetryService.GetDefaultSession().PostEvent(telemetryEvent);
+            }
             return Task.CompletedTask;
         }
 
+        private string GetTypespecVersion()
+        {
+            try
+            {
+                ProcessStartInfo processStartInfo = new()
```

Comment:
I think this will have problem like when the tsp is only installed locally. And what if VS is installed on Linux/Mac?